src/lib/optimizerTabController.js

```diff
@@ -94,11 +94,27 @@ export const OptimizerTabController = {
       if (event.data && fieldValues.characterId) {
         const character = DB.getCharacterById(fieldValues.characterId)
 
-        if (data.id) {
+        if (!data.id) {
+          window.optimizerGrid.current.api.deselectAll()
+        }
+
+        if (character && data.id) {
           const rowId = data.id
           const build = OptimizerTabController.calculateRelicsFromId(rowId)
           window.setOptimizerBuild(build)
-          selectRow(data.id)// feels kinda hacky but it works ¯\_(ツ)_/¯
+
+          // Find the row by its string ID and select it
+          const rowNode = window.optimizerGrid.current.api.getRowNode(String(data.id))
+          if (rowNode) {
+            const currentPinned = window.optimizerGrid.current.api.pinnedRowModel.pinnedTopRows.map((x) => x.data)
+
+            if (String(currentPinned[0].id) == String(rowNode.data.id)) {
+              // The currently equipped top row shouldn't correspond to an optimizer row, deselect
+              window.optimizerGrid.current.api.deselectAll()
```

Comment:
after equipping an optimizer build the top row will have a corresponding optimizer row

Reply:
I wanted to disconnect them because the top row can be affected by other tabs (un/equipping stuff on the relics tab) while the optimizer rows wont change